chttp/cookieauth.go

```diff
@@ -89,7 +89,21 @@ func (a *CookieAuth) RoundTrip(req *http.Request) (*http.Response, error) {
 	if err := a.authenticate(req); err != nil {
 		return nil, err
 	}
-	return a.transport.RoundTrip(req)
+
+	res, err := a.transport.RoundTrip(req)
+	if err != nil {
+		return res, err
+	}
+
+	if res != nil && res.StatusCode == http.StatusUnauthorized {
+		if a.Cookie() != nil {
+			a.client.Jar = nil
+			a.setCookieJar()
+		}
+		// FIXME: do we want to re-try the request here?  That seems like a
+		// violition of the interface intention.
```

Comment:
typo

Reply:
Dropped the comment as I realised that was definitely an approach to avoid.